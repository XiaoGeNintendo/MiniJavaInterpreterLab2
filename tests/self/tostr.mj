class A{
}
class D extends A{}
class B extends A{
    string x;
    B(string x){
        this.x=x;
    }
    string to_string(){
        return "114514"+x;
    }
}
class C extends B{
    C(string x){
        super(x);
    }
}
int main(){
    println(new A());
    println(new B("!"));
    println(new C("!!"));
    println((A)null);
    println((B)null);
    println((C)null);
    A a=null;
    B b=null;
    C c=null;
    println(a);
    println(b);
    println(c);
    a=new A();
    println(a);
    a=new B("?");
    println(a);
    a=new C("??");
    println(a);
    a=new D();
    println(a);
    return 0;
}