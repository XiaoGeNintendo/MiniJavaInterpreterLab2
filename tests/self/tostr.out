A
114514!
114514!!
null
null
null
null
null
null
A
B
C
D
114514??
Process exits with the code 0.
